File: Bsearch.cpp
#include <iostream>

using namespace std;

int Bsearch(int n, int x[40], int item);

int main()
{
   int a[40],n,item, pos;
   cout<<"enter number of elements in data"<<endl;
   cin>>n;
   cout<<"enter data(in ascending order)"<<endl;
   for(int j=0;j<n;j++)
        cin>>a[j];
   cout<<"enter item to look for"<<endl;
   cin>>item;
   pos = Bsearch(n,a,item);
   cout<<"the position of your element is:"<<pos+1<<endl;

   return 0;
}
int Bsearch(int n, int x[40], int item)
{
    int position=0,mid,u=0,f,flag=0;
    f=n-1;
    mid=(u+f)/2;
    for(mid=n/2;mid>=0&&mid<=n;mid=(u+f)/2)
    {
        if(item==x[mid])
        {
            position=mid;
            flag=1;
            break;
        }
    else if(item>x[mid])
        {
            u=mid+1;
        }
    else
        {
           f=mid-1;
        }
    }
    if(flag==0)
        cout<<"iten is not a part of data"<<endl;
    return position;
}

File: bsearch.py
list = [int(item) for item in input("Enter list (seperated by space): ").split()]
num = int(input("Enter element to be found: "))

def binary_search(l, x):
    l.sort()
    mid = int(len(l)/2)
    flag = "not found"
    while True:
        if l[mid]>x:
            if mid/2 < 1:
                break
            mid=int(mid/2)
        elif l[mid]<x:
            mid+=int(mid/2)
            if mid > len(l)-1:
                break
        elif l[mid]==x:
            flag = f"found at {mid}"
            break
        else:
            print(l[mid])
            print(x)
    return flag

print(binary_search(list, num))
